kratos/processes/find_global_nodal_elemental_neighbours_process.h

```diff
@@ -0,0 +1,449 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Riccardo Rossi
+//
+
+
+#if !defined(KRATOS_FIND_GLOBAL_NODAL_ELEMENTAL_NEIGHBOURS_PROCESS_H_INCLUDED )
+#define  KRATOS_FIND_GLOBAL_NODAL_ELEMENTAL_NEIGHBOURS_PROCESS_H_INCLUDED
+
+
+
+// System includes
+#include <string>
+#include <iostream>
+
+
+// External includes
+
+
+// Project includes
+#include "includes/define.h"
+#include "processes/process.h"
+#include "includes/node.h"
+#include "includes/element.h"
+#include "includes/model_part.h"
+#include "includes/global_pointer_variables.h"
+#include "utilities/communication_coloring_utilities.h"
+#include "utilities/pointer_communicator.h"
+#include "includes/mpi_serializer.h"
+#include "utilities/compute_neighbour_list_functor.h"
+#include "utilities/global_pointer_utilities.h"
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+typedef  ModelPart::NodesContainerType NodesContainerType;
+typedef  ModelPart::ElementsContainerType ElementsContainerType;
+
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// Short class definition.
+/** Detail class definition.
+*/
+class FindGlobalNodalElementalNeighboursProcess
+    : public Process
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of FindGlobalNodalElementalNeighboursProcess
+    KRATOS_CLASS_POINTER_DEFINITION(FindGlobalNodalElementalNeighboursProcess);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    /// the better the guess for the quantities above the less memory occupied and the fastest the algorithm
+    FindGlobalNodalElementalNeighboursProcess(const DataCommunicator& rComm, 
+                                              ModelPart& model_part)
+        : mrComm(rComm),mr_model_part(model_part)
+    {
+    }
+
+    /// Destructor.
+    ~FindGlobalNodalElementalNeighboursProcess() override
+    {
+    }
+
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    void operator()()
+    {
+        Execute();
+    }
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    void Execute() override
+    {
+        unsigned int current_rank = mrComm.Rank();
+        auto& rNodes = mr_model_part.Nodes();
+
+        //first of all the neighbour nodes and elements array are initialized to the guessed size
+        //and empties the old entries
+        for(NodesContainerType::iterator in = rNodes.begin(); in!=rNodes.end(); in++)
```

Comment:
```suggestion
        for (auto& rNode : rNodes)
```
you sure this should also loop the ghosts?

Reply:
yes, this should be on everything.

about your suggestion, both forms are legit ... so i don't really see a reason for changing this (nor, not to really)

in any case i took the occasion to parallelize the loops